### PR TITLE
perf test

### PR DESCRIPTION
**Pull request by DIBL**

### CODE REVIEW

File: src/main/java/com/Elessar/app/Main.java [hunk 1]
```diff
@@ -1,49 +1,101 @@
...
+     *
+     * @param args [0] username for user1
+     *             [1] port number for user1
+     *             [2] username for user2
+     *             [3] port number for user2
+     *             [4] total number of messages to send
+     */
+    public static void main(String[] args) throws Exception {
+        final String serverURL = new StringBuilder().append("http://127.0.0.1:9000").toString();
+        final MetricManager metricManager = new MetricManager("ClientMetric", 0);
```
Comment (hypnotichen) [inline src/main/java/com/Elessar/app/Main.java]: You need to choose a realistic maxEntryCount, especially when we are doing perf test here. Writting to disk too frequently will negatively impact overall performance

File: src/main/java/com/Elessar/app/Main.java [hunk 2]
```diff
@@ -1,49 +1,101 @@
...
+import java.util.Random;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
 
 /**
- * Created by Hans on 1/7/19.
+ * Created by Hans on 2/27/19.
  */
-
 public class Main {
```
Comment (hypnotichen) [inline src/main/java/com/Elessar/app/Main.java]: Give this class a better name...

File: src/main/java/com/Elessar/app/Main.java [hunk 3]
```diff
@@ -1,49 +1,101 @@
...
-            sendPersonData.build().writeTo(os);
-            try (InputStream is = new ByteArrayInputStream(os.toByteArray())) {
-                final Person receivePersonData = Person.parseFrom(is);
-                System.out.printf("receive and send data is equal: %b\n", sendPersonData.equals(receivePersonData));
-                System.out.println(receivePersonData.toString());
-                System.out.println(sendPersonData.toString());
+        while (messageCount > 0) {
+            int msgCount1 = r.nextInt(5) + 1;
+            while (messageCount > 0 && msgCount1 > 0) {
+                client1.sendMessage(username1, username2, getRandomStr(10));
```
Comment (hypnotichen) [inline src/main/java/com/Elessar/app/Main.java]: Maybe we want to do some timing here as well, like avg / p90 time of how long does a client request get a response.
Also we want to keep track how many requests failed. If failure rate is high then result of the perf test is not reliable

Comment (DIBL) [reply src/main/java/com/Elessar/app/Main.java]: > Maybe we want to do some timing here as well, like avg / p90 time of how long does a client request get a response.
Why not use metric measured from MyClient for this performance?

Comment (hypnotichen) [reply src/main/java/com/Elessar/app/Main.java]: You are right... I just forgot we had metrics in MyClient. 
Ok we don't need metrics in perfMain, just keep track of error is enough. My bad.

**Pull request merged**